pandas/core/indexes/datetimelike.py

```diff
@@ -209,6 +209,16 @@ def ceil(self, freq):
 class DatetimeIndexOpsMixin(object):
     """ common ops mixin to support a unified interface datetimelike Index """
 
+    @property
+    def base(self):
+        """ return the base object if the memory of the underlying data is
+        shared
+        """
+        # override deprecated property in IndexOpsMixin, as we still need
```

Comment:
your explanation doesn't make sense, we are deprecating these no? you need to change the way its accessed in the code to remove the warnings.

Reply:
The explanation perfectly makes sense, but you don't like the way I solved it I suppose.

`.base` is used in two placed:

- `Block.is_view`: this I can override in `DatetimeTzBlock` to check `self.values.values.base`
- `concatenate_join_units`: https://github.com/pandas-dev/pandas/blob/ede11af8d02a4ac37ed866593839024b941a8086/pandas/core/internals.py#L5544  Not fully sure about this one what to do here.

Reply:
my point is this method should not exist (as you are ddeprecateding), and would rather have you fix the usage

Reply:
As I said, I agree with that. Do you have a suggestion for the second case? (inside `concatenate_join_units`) In https://github.com/pandas-dev/pandas/pull/20745 I need to touch the same code, and checked if the values have a `base` attribute, but that also feels hacky

Reply:
yes, see my comment there, only check base if its an ndarray